FWCore/Framework/test/stubs/TestStreamProducers.cc

```diff
@@ -219,23 +200,22 @@ namespace edmtest {
                                            LuminosityBlockContext const* iLBContext) {
         ++m_count;
         auto pCache = iLBContext->luminosityBlock();
-        if (pCache->lumi != 1) {
+        if (pCache->run != iLB.luminosityBlockAuxiliary().run() ||
+            pCache->lumi != iLB.luminosityBlockAuxiliary().luminosityBlock()) {
           throw cms::Exception("end out of sequence")
               << "globalEndLuminosityBlock seen before globalBeginLuminosityBlock in LuminosityBlock"
               << iLB.luminosityBlock();
         }
-        gel = true;
-        gbl = false;
+        pCache->run = 0;
+        pCache->lumi = 0;
         if (iLBContext->luminosityBlock()->value != cvalue_) {
           throw cms::Exception("cache value")
               << iLBContext->luminosityBlock()->value << " but it was supposed to be " << cvalue_;
         }
       }
 
-      static void endLuminosityBlock(edm::Run const&, edm::EventSetup const&, LuminosityBlockContext const*) {
-        el = true;
-        bl = false;
-        if (gel) {
+      void endLuminosityBlock(edm::LuminosityBlock const& iLB, edm::EventSetup const&) override {
+        if (luminosityBlockCache()->run != iLB.luminosityBlockAuxiliary().run()) {
```

Comment:
Should this check also the LuminosityBlock number?

Reply:
Done.